app/src/main/java/com/babylon/wallet/android/presentation/settings/linkedconnectors/LinkedConnectorsScreen.kt

```diff
@@ -228,48 +268,104 @@ private fun ActiveLinkedConnectorsListContent(
+                overflow = TextOverflow.Ellipsis
             )
+            IconButton(
+                onClick = { onRenameConnectorClick(activeLinkedConnector) }
+            ) {
+                Icon(
+                    painter = painterResource(id = com.babylon.wallet.android.designsystem.R.drawable.ic_account_label),
+                    contentDescription = null,
+                    tint = RadixTheme.colors.gray1
+                )
+            }
             IconButton(onClick = {
                 onDeleteConnectorClick(activeLinkedConnector.id)
             }) {
                 Icon(
-                    painter = painterResource(id = R.drawable.ic_delete_24),
+                    painter = painterResource(id = com.babylon.wallet.android.designsystem.R.drawable.ic_delete_outline),
                     contentDescription = null,
                     tint = RadixTheme.colors.gray1
                 )
             }
         }
-        HorizontalDivider(Modifier.fillMaxWidth(), color = RadixTheme.colors.gray4)
     }
 }
 
-@Preview(showBackground = true)
 @Composable
-fun LinkedConnectorsContentWithoutActiveLinkedConnectorsPreview() {
-    RadixWalletTheme {
-        LinkedConnectorsContent(
-            activeLinkedConnectorsList = persistentListOf(),
-            onLinkNewConnectorClick = {},
-            isAddingNewLinkConnectorInProgress = false,
-            onBackClick = {},
-            onDeleteConnectorClick = {}
+private fun RenameActiveLinkedConnectorSheet(
+    input: LinkedConnectorsUiState.RenameConnectorInput,
+    onNewNameChange: (String) -> Unit,
+    onUpdateNameClick: () -> Unit,
+    onDismiss: () -> Unit
+) {
+    val inputFocusRequester = remember { FocusRequester() }
+
+    LaunchedEffect(Unit) {
+        inputFocusRequester.requestFocus()
+    }
+
+    BottomSheetDialogWrapper(
+        addScrim = true,
+        showDragHandle = true,
+        onDismiss = onDismiss
+    ) {
+        Column(
+            modifier = Modifier.verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.Center,
+        ) {
+            Text(
+                modifier = Modifier.fillMaxWidth(),
+                text = stringResource(id = R.string.linkedConnectors_renameConnector_title),
+                style = RadixTheme.typography.title,
+                color = RadixTheme.colors.gray1,
+                textAlign = TextAlign.Center
+            )
+            Spacer(modifier = Modifier.height(RadixTheme.dimensions.paddingSemiLarge))
+            Text(
+                modifier = Modifier.fillMaxWidth(),
+                text = stringResource(id = R.string.linkedConnectors_renameConnector_subtitle),
+                style = RadixTheme.typography.body1Regular,
+                color = RadixTheme.colors.gray1,
+                textAlign = TextAlign.Center
+            )
+            Spacer(modifier = Modifier.height(RadixTheme.dimensions.paddingLarge))
+            RadixTextField(
```

Comment:
It would be nice to make the name input start with uppercase:
`keyboardOptions = KeyboardOptions.Default.copy(capitalization = KeyboardCapitalization.Words)`